### DISCUSSION TITLE
Brackets.next 2.0.0 release features discussion thread

### DISCUSSION THREAD

**Discussion by abose** [brackets-cont/brackets#23] [Feature Request]
This thread is to identify features to be shipped as the next releases of brackets. Please comment and upvote features in this thread.

* [See this project for details on the November release](https://github.com/brackets-cont/brackets/projects/2?add_cards_query=is%3Aopen).

**Comment by abose** [comment]
Should we switch brackets shell to electron?

**Reply by abose** [reply]
More details on the electron port can be found here: https://github.com/brackets-cont/brackets/issues/26

**Comment by Cyberpwince** [comment]
I use bracket for php development

**Reply by abose** [reply]
Hi @Cyberpwince Thanks for the note. Can you give more details on the workflow you use for PHP in brackets if possible?

**Comment by jhcarlos8888** [comment]
Good evening I use php to modify projects made in drupal and files of web pages that I have. I use the PHP language too much and I think it is very necessary that this great editor has support and extensions to program in PHP. Please do not leave this great language in the trash because it really is very necessary for and many people use it to work daily

**Reply by abose** [reply]
Thanks for the input! We are still investigating ways to get it working, Will update before the release.

**Comment by nyanko-oss** [comment]
i also use bracket for edit to php and also php, css projects. We need php extension, don't remove please!

**Comment by satovey** [comment]
If you're going to do a PHP extension, perhaps building an extension that is similar to something like Desktop PHP is in order.

Php Desktop allows the user to run php code in the /www sub folder.

When the App.exe is executed, it calls chromium CEF which then looks for the index.html or index.php file under the /www folder and executes the the web page.

The folder structure of PHP Desktop on Windows is as follows;
PHP Desktop
	-locales
	-php
	-webcache
	-www
App.exe
cef.pak
cef_100_percent.pak
cef_200_percent.pak
cef_extensions.pak
chrome_elf.dll
d3dcompiler_43.dll
d3dcompiler_47.dll
debug.log
devtools_resources.pak
icudtl.dat
libcef.dll
libEGL.dll
libGLESv2.dll
license.txt
natives_blob.bin
settings.json
snapshot_blob.bin
widevinecdmadapter.dll

There are versions for other languages as well called; Python Desktop, Ruby Desktop and perl-desktop and given that each of those scripting languages have an internal web server to fun code from; version could be built for NodeJS as well.

While PHP Desktop only incorporates a single version of the php language; the language is easily upgradeable by simply updating php in the /php folder. However, a little tweaking and you could allow for multiple versions of php by placing the versions under the /php folder as follows;

PHP-Desktop
	-locales
	-php
		-php4
		-php5
		-php7
	-webcache
	-www

Another nice thing about this is that you can use the PHP interpreter to debug the program and output the errors and warnings in a prompt window. 

If Brackets chose the scripting language based on the file extension, and then passed that parameter onto the CEF which then ran the engines internal web server, you could add the four main scripting engines; PHP, Python, Ruby on Rails and NodeJS into the sub folder as well as support any other scripting language that has the same features as the PPRuNo four do.

The structure might look like as follows
Brackets
	-ScriptEngines
		-locales
		-php
		-python
		-ruby
		-nodejs
		-webcache
		-www

If you can then figure out how to cross compile the code, you would be able to build cross platform apps that function similar to native apps; using those scripting engines.

**Reply by abose** [reply]
Thanks for the detailed workflow notes :) 
Tagging @charlypa to follow up

**Reply by charlypa** [reply]
@satovey Thanks for your recommendation. Right now, we're able to fix the issue. We will go through it in detail while we start with electron migration.
https://github.com/brackets-cont/brackets/issues/26

**Reply by satovey** [reply]
You're welcome!

**Reply by abose** [reply]
https://github.com/brackets-cont/brackets/issues/88

**Comment by abose** [comment]
@Cyberpwince @jhcarlos8888 @nyanko-oss @satovey 
We have reinstated the PHP extension back in Brackets. PHP support will be available as standard in brackets 2.0 release.

**Comment by abose** [comment]
Fixed in pre-release builds. Closing
